Lib/fontTools/pens/basePen.py

```diff
@@ -350,20 +374,24 @@ def decomposeSuperBezierSegment(points):
 			factor = j / nDivisions
 			temp1 = points[i-1]
 			temp2 = points[i-2]
-			temp = (temp2[0] + factor * (temp1[0] - temp2[0]),
-					temp2[1] + factor * (temp1[1] - temp2[1]))
+			temp = (
+				temp2[0] + factor * (temp1[0] - temp2[0]),
+				temp2[1] + factor * (temp1[1] - temp2[1]),
+			)
 			if pt2 is None:
 				pt2 = temp
 			else:
-				pt3 =  (0.5 * (pt2[0] + temp[0]),
-						0.5 * (pt2[1] + temp[1]))
+				pt3 =  (
+					0.5 * (pt2[0] + temp[0]),
+					0.5 * (pt2[1] + temp[1]),
+				)
 				bezierSegments.append((pt1, pt2, pt3))
 				pt1, pt2, pt3 = temp, None, None
 	bezierSegments.append((pt1, points[-2], points[-1]))
 	return bezierSegments
 
 
-def decomposeQuadraticSegment(points):
+def decomposeQuadraticSegment(points: List[Union[PenPoint, None]]) -> List[Tuple[PenPoint, PenPoint]]:
```

Comment:
You can `from __future__ import annotations` at the top and then rewrite this as `def decomposeQuadraticSegment(points: 
list[PenPoint | None]) -> list[tuple[PenPoint, PenPoint]]:` and save yourself the `List`, `Union` and `Tuple` imports :)

Reply:
Thanks! I wasn't sure if Python 3.7 can deal with that notation.

Reply:
The annotations import was added in 3.7. You still need to import the typings for type aliases though... until we have 3.10 minimum.

Reply:
@anthrotype actually, I'm wondering about the `points` argument, Is `None` really possible? The docstring says the last point must be an on-curve point, implying that it cannot be `None`?

Reply:
Yes the last point in the list (and only the last) can be None, which indicates a closed TrueType contour containing only off-curve points and no on-curves..

Reply:
Yeah, but in this instance? It's only called from https://github.com/fonttools/fonttools/blob/16884008daa45a04cfe9a526aff767053e24b09a/Lib/fontTools/pens/basePen.py#L348 where the last point is taken care of?

Reply:
Good point you are right the caller handles that case already, if this function expects all points to be non-None that we can type it without the Optional, maybe add an assertion too